camayoc/qpc_models.py

```diff
@@ -168,6 +169,7 @@ def __init__(
         username=None,
         password=None,
         ssh_keyfile=None,
+        token=None,
```

Comment:
I think changes like that should be accompanied by configuration spec changes right away.

In `camayoc/types/settings.py`, add a new class:

```
class OpenshiftTokenCredentialOptions(BaseModel):
    name: str
    type: Literal["openshift"]
    auth_token: str
```

And a little below:
```
CredentialOptions = Annotated[
    Union[
        PlainNetworkCredentialOptions,
        SSHNetworkCredentialOptions,
        SatelliteCredentialOptions,
        VCenterCredentialOptions,
        OpenshiftTokenCredentialOptions,
    ],
    Field(discriminator="type"),
]
```

I would go with OpenshiftToken because we already know that Openshift username + password are going to be a thing.